pytest_fixtures/component/repository.py

```diff
@@ -59,3 +59,138 @@ def module_repository(os_path, module_product, module_target_sat):
     repo = module_target_sat.api.Repository(product=module_product, url=os_path).create()
     call_entity_method_with_timeout(module_target_sat.api.Repository(id=repo.id).sync, timeout=3600)
     return repo
+
+
+def _simplify_repos(request, repos):
+    """This is a helper function that transforms repos_collection related fixture parameters into
+    a list that can be passed to robottelo.host_helpers.RepositoryMixins.RepositoryCollection
+    class constructor arguments. It can create multiple repos with distro.
+
+    E.g: the parameters list to repos_collection fixture is:
+    [{
+        'distro': DISTRO_RHEL7,
+        'SatelliteToolsRepository': {},
+        'YumRepository': [{'url': settings.repos.yum_0.url},{'url': settings.repos.yum_6.url}]
+    }]
+
+    The fixture removes distro from the dict above and translates the remaining dictionary into
+    [
+        {'SatelliteToolsRepository': {}},
+        {'YumRepository': {'url': settings.repos.yum_0.url}},
+        {'YumRepository': {'url': settings.repos.yum_6.url}}
+    ]
+    Then the fixtures loop over it to create multiple repositories.
+
+    :returns: The tuple of distro of repositories(if given) and simplified repos
+    """
+    _repos = []
+    repo_distro = None
+    # Iterating over repository thats requested more than once
+    for repo_name, repo_options in repos.items():
+        if isinstance(repo_options, list):
```

Comment:
Might want to look for iterable here instead of just `list`

Reply:
No that wont work because a dictionary with only one repository to create with data, lets say `{'url': 'settings.repos.yum_6.url'}` is also iterable.